buildenv/jenkins/JenkinsfileBase

```diff
@@ -108,34 +108,63 @@ def setupEnv() {
 def setupParallelEnv() {
 	stage('setupParallelEnv') {
 		timestamps{
-			setupEnv()
+			setupEnv()		
+			int iterate = 1
+			if (params.ITERATIONS) {
+				iterate = "${params.ITERATIONS}".toInteger()
+			}
```

Comment:
Please set variable in  setupEnv()
I think we do not need both  `repeatAmount` and `iterate`. One should be enough.

Reply:
Both variables are needed, because they have different functions. The variable iterate is the number of iterations as passed by the user. The variable repeatAmount is the number of times the loop runs; this is equal to ITERATIONS if it's greater than one, and the number of subdirectories if ITERATIONS is equal to one.

Reply:
I think we can do something like below:
```
repeatAmount = "${params.ITERATIONS}".toInteger()
if (repeatAmount == 1) {
   // get testSubDirSize and set it
   repeatAmount = testSubDirSize
}
```

Reply:
I just tried to move the variable in setupEnv(), and it didn't work. I also noticed in setupEnv() that there are no ints in there. The function is also used to setup situations both parallel as well as not, whereas iterate is only used for parallel situations. This is also a local variable that was there previously, so I'm not really sure if it's necessary to move it.

Reply:
`iterations` is not only used in parallel situations. We also use it in `runTest()`:
https://github.com/AdoptOpenJDK/openjdk-tests/blob/b3ce67da9cf6c678eaffbbabb8abeb763f8482ed/buildenv/jenkins/JenkinsfileBase#L327

So it is better to set it once.

Reply:
I'm not completely sure how to do this. I tried yesterday by defining the int in the setupEnv(), and there are no local variables being declared there. Should I make `iterations` a global variable, or is there some other way to do this?

Reply:
We can set it in `env`.